src/Netscript/NetscriptHelpers.tsx

```diff
@@ -297,17 +297,18 @@ function validateHGWOptions(ctx: NetscriptContext, opts: unknown): CompleteHGWOp
     return result;
   }
   if (typeof opts !== "object") {
-    throw errorMessage(ctx, `BasicHGWOptions must be an object if specified, was ${String(opts)}`);
+    // eslint-disable-next-line @typescript-eslint/no-base-to-string
+    throw errorMessage(ctx, `BasicHGWOptions must be an object if specified, was ${opts}`);
   }
   // Safe assertion since threadOrOption type has been narrowed to a non-null object
   const options = opts as Unknownify<CompleteHGWOptions>;
   result.stock = !!options.stock;
   result.additionalMsec = number(ctx, "opts.additionalMsec", options.additionalMsec ?? 0);
   if (result.additionalMsec < 0) {
-    throw errorMessage(ctx, `additionalMsec must be non-negative, got ${String(options.additionalMsec)}`);
+    throw errorMessage(ctx, `additionalMsec must be non-negative, got ${options.additionalMsec}`);
```

Comment:
Did you mean result.additionalMsec?

options.additionalMsec is the unknown input, result.additionalMsec is the number. If you suspect it's a non-numeric, nullish, and non-string value then it's best to JSON.stringify it rather than string interpolate

Reply:
Shouldn't it be stringified then through something that can survive unknown values? JSON.stringify or equivalent? In this case only since it's a property of a structured object.

Reply:
Using JSON.stringify is useful when we try to print unknown data, and that data may be an object. In this case, `options.additionalMsec` is guaranteed to be a negative number. Theoretically, when we have that information, it may be better to just use the "type-narrowed" variable (`result.additionalMsec`). The usage of the original variable is kind of a "default action". IMO, printing the original variable is the better action by default. We can use JSON.stringify when it's useful.

Reply:
My comment above was related to that. Since this particular check knows that it's already a negative number you can use the error with result.additionalMsec to carry over the correct type, rather than using the more "DRY pattern" that then requires a type cast.